error/s2n_errno.c

```diff
@@ -275,6 +275,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_LIBCRYPTO_VERSION_NUMBER_MISMATCH, "The libcrypto major version number seen at compile-time is different from the major version number seen at run-time") \
     ERR_ENTRY(S2N_ERR_LIBCRYPTO_VERSION_NAME_MISMATCH, "The libcrypto major version name seen at compile-time is different from the major version name seen at run-time") \
     ERR_ENTRY(S2N_ERR_OSSL_PROVIDER, "Failed to load or unload an openssl provider") \
+    ERR_ENTRY(S2N_ERR_INVALID_CHACHA20_BOOSTING_STATE, "Server and/or client did not indicate chacha20 boosting support") \
     ERR_ENTRY(S2N_ERR_CERT_OWNERSHIP, "The ownership of the certificate chain is incompatible with the operation") \
```

Comment:
Which error would you recommend then?

Reply:
S2N_ERR_INVALID_STATE would probably be my choice, but it doesn't particularly matter. It would matter if you wanted to check for it in tests or surface it to customers.

If you do want to keep this error, "S2N_ERR_CHACHA20_BOOSTING_UNSUPPORTED" would probably be more accurate. Also, for the message, "chacha20 boosting support" doesn't really make sense for clients. Servers support chacha20 boosting, clients just prefer chacah20.